text/0000-glimmer-components.md

```diff
@@ -508,47 +553,52 @@ cycle in a number of ways:
   `didUpdateAttrs` do _not_ have opposing bookends is inconsistent with this
   pattern.
 
-* Hooks that run immediately _before_ a render such as `didUpdateAttrs` are not
-  particularly useful, because the only functionality they have which
-  post-render methods do not is to modify a bound value without triggering a
-  rerender:
+* Hooks which are used to manipulate elements or the DOM in general can be
+  removed in favor of element modifiers, which are discussed in detail in the
+  next section.
 
-  ```js
-  willUpdateAttrs() {
-    if (this.someBoolArg) {
-      this.set('someBoundValue', true);
-    }
-  }
-  ```
+Based on these considerations, we can reduce these hooks to one per major phase
+of the component's lifecycle: the `constructor`, `onUpdate`, and `onDestroy`.
 
-  These types of values can almost always be better expressed using computed
-  properties or tracked getters, which will update when accessed during render,
-  preventing rerenders. Conversely, much imperative logic _must_ be done after a
-  render (manipulating an element directly, for instance), so post render hooks
-  are the most sensible.
+#### `constructor`
 
-* The element based methods can be removed in favor of element modifiers,
-  which are discussed later on.
+The `constructor` method can be used for initial setup of the component. This
+effectively replaces `init`, and allows users to setup state before _any_
+renders occur. It has the following timing semantics:
 
-Based on these considerations, we can reduce these hooks to one per major phase
-of the component's lifecycle:
+* **Always**
+  * called when a component is created
+  * called _before_ any child components are created
 
-The **`constructor`** can be used for initial setup of the component. This
-effectively replaces `init`, and allows users to setup state before _any_
-renders occur.
+#### `onUpdate`
+
+This hook runs after the component's `args` have been updated, allowing users to
+run any code they need in order to react to changes external to the component.
+This is particularly useful for running asynchronous code based on the value of
+an argument, such as `fetch` requests or ember-concurrency style tasks. It has
+the following timing semantics:
 
-**`didCreate`** runs after the first full render of the component, once the
-component has been fully created. It is closest to `didInsertElement` in classic
-components, but has been renamed since components no longer have an `element`
-property.
+* **Always**
+  * called when `args` have changed
+  * called _before_ any child component `onUpdate` hooks
+* **May or May Not**
```

Comment:
Should also state that it may be called even if the contents are stable (e.g. we _guarantee_ it will be called if they are updated, but it may be called if they aren't too...)